File: c++/arrays/examples_of_array_elements.cpp
#include <iostream>

int main()
{
  int numbers[] = {2, 4, 6};

  std::cout << "numbers[0]: " << numbers[0] << std::endl;
  std::cout << "numbers[1]: " << numbers[1] << std::endl;
  std::cout << "numbers[2]: " << numbers[2] << std::endl << std::endl;

  double values[] = {9.5, 8.5, 7.5};

  std::cout << "values[0]: " << values[0] << std::endl;
  std::cout << "values[1]: " << values[1] << std::endl;
  std::cout << "values[2]: " << values[2] << std::endl << std::endl;

  char letters[] = {'a', 'b', 'c'};

  std::cout << "letters[0]: " << letters[0] << std::endl;
  std::cout << "letters[1]: " << letters[1] << std::endl;
  std::cout << "letters[2]: " << letters[2] << std::endl << std::endl;

  const char* words[] = {"first", "two", "last"};

  std::cout << "words[0]: " << words[0] << std::endl;
  std::cout << "words[1]: " << words[1] << std::endl;
  std::cout << "words[2]: " << words[2] << std::endl << std::endl;

  enum rating
  {
    excellent,
    high,
    medium,
    low,
    substandard
  };

  rating evaluations[] = {low, medium, high};

  std::cout << "evaluations[0]: " << evaluations[0] << std::endl;
  std::cout << "evaluations[1]: " << evaluations[1] << std::endl;
  std::cout << "evaluations[2]: " << evaluations[2] << std::endl << std::endl;

  struct Animal
  {
    const char* breed;
    Animal(const char* breed) : breed{breed} {}
  } cow("Cattle"), cat("Siamse"), dog("Samoyed"), hamster("Syrian"), rabbit("Tan"), parrot("Amazon");

  Animal menagerie[] = {cow, hamster, rabbit};

  std::cout << "menagerie[0]: " << menagerie[0].breed << std::endl;
  std::cout << "menagerie[1]: " << menagerie[1].breed << std::endl;
  std::cout << "menagerie[2]: " << menagerie[2].breed << std::endl << std::endl;
}

File: pascal/arrays/examples_of_array_elements.pas
program ExamplesOfArrayElements;

type
  rating = (
    excellent,
    high,
    medium,
    low,
    substandard
  );

  Animal = record
    breed: string;
  end;

var
  numbers: array[0..2] of integer = (2, 4, 6);
  values: array[1..3] of real = (9.5, 8.5, 7.5);
  letters: array[-3..-1] of char = ('a', 'b', 'c');
  words: array['a'..'c'] of string = ('first', 'two', 'last');
  evaluations: array[0..2] of rating = (low, medium, high);
  menagerie: array[0..2] of Animal = ((breed: 'Cattle'), (breed: 'Syrian'), (breed: 'Tan'));

begin
  writeln('numbers[0]: ', numbers[0]);
  writeln('numbers[1]: ', numbers[1]);
  writeln('numbers[2]: ', numbers[2]);
  writeln();

  writeln('values[1]: ', values[1]);
  writeln('values[2]: ', values[2]);
  writeln('values[3]: ', values[3]);
  writeln();

  writeln('letters[-3]: ', letters[-3]);
  writeln('letters[-2]: ', letters[-2]);
  writeln('letters[-1]: ', letters[-1]);
  writeln();

  writeln('words[''a'']: ', words['a']);
  writeln('words[''b'']: ', words['b']);
  writeln('words[''c'']: ', words['c']);
  writeln();

  writeln('evaluations[0]: ', evaluations[0]);
  writeln('evaluations[1]: ', evaluations[1]);
  writeln('evaluations[2]: ', evaluations[2]);
  writeln();

  writeln('menagerie[0]: ', menagerie[0].breed);
  writeln('menagerie[1]: ', menagerie[1].breed);
  writeln('menagerie[2]: ', menagerie[2].breed);
  writeln();
end.
